lib/screens/v2/candidate_generator/elevator/elevator_closures.ex

```diff
@@ -0,0 +1,85 @@
+defmodule Screens.V2.CandidateGenerator.Elevator.ElevatorClosures do
+  @moduledoc false
+
+  require Logger
+
+  alias Screens.Alerts.Alert
+  alias Screens.Facilities.Facility
+  alias Screens.Routes.Route
+  alias Screens.Stops.Stop
+  alias Screens.V2.WidgetInstance.ElevatorClosures
+  alias ScreensConfig.Screen
+  alias ScreensConfig.V2.Elevator
+
+  @stop Application.compile_env(:screens, [__MODULE__, :stop_module], Stop)
+  @facility Application.compile_env(:screens, [__MODULE__, :facility_module], Facility)
+  @alert Application.compile_env(:screens, [__MODULE__, :alert_module], Alert)
+  @route Application.compile_env(:screens, [__MODULE__, :route_module], Route)
+
+  def elevator_status_instances(
+        %Screen{
+          app_params: %Elevator{
+            elevator_id: elevator_id
+          }
+        } = config,
+        now \\ DateTime.utc_now()
+      ) do
+    with {:ok, %Stop{id: stop_id}} <- @facility.fetch_stop_for_facility(elevator_id),
+         {:ok, location_context} <- @stop.fetch_location_context(Elevator, stop_id, now),
+         {:ok, parent_station_map} <- @stop.fetch_parent_station_name_map(),
+         {:ok, alerts} <- @alert.fetch_elevator_alerts_with_facilities() do
+      elevator_closures = relevant_alerts(alerts)
+      routes_map = get_routes_map(elevator_closures, stop_id)
+
+      [
+        %ElevatorClosures{
+          alerts: elevator_closures,
+          location_context: location_context,
+          screen: config,
+          now: now,
+          station_id_to_name: parent_station_map,
+          station_id_to_routes: routes_map
```

Comment:
One piece of consolidation I just thought of is making these two values a single 
`%{String.t() => %{routes: list(String.t()), name: String.t()}`. 

I know that doesn't really address your concern completely though. A specific barrier I tend to put up (which I don't think is foreign to others) is API work goes in the CG. That's the reason for any of this living in the CG. As for why the serializer needs these maps, I mostly just don't see an elegant way to add these extra values to the specific alert. I suppose I could create a new struct like `AlertWithContext` where there is the alert info we need plus the station info. What do you think?

Reply:
I guess my thinking was, values don't need to be passed from the candidate generator to the widget module at all if they're both produced _and_ consumed entirely within one or the other. The split between "CG" and "widget" seems predicated on the idea that maybe someday there will be multiple generators that can produce the same widget, but currently there aren't and I don't think we foresee that happening, so anywhere we draw this line is going to be arbitrary. Given that, why not choose a line that minimizes "surface area"?

As an example, what if the widget module was _just_ this (and the protocol implementation):

```elixir
defmodule Screens.V2.WidgetInstance.ElevatorClosures do
  defstruct ~w[id in_station_alerts outside_alerts]a

  defmodule Alert do
    defstruct ~w[station_name routes alert_id elevator_name elevator_id description header_text]a
  end
  
  def serialize(t), do: t
end
```

I feel like this would make the full-stack behavior of the widget a lot easier to understand, as there isn't this extra intermediate form of it — "what you generate is what you get". And it would also have benefits for test coverage of the "complete" widget logic.

At the same time, I recognize this is a significant departure from how we've built generators and widgets until now, and there could be other aspects or downsides to it, which an unrelated PR might not be the best venue to discuss. So I present this more as a thought-provoking exercise than a change I think we need to make here and now.

Reply:
Hmm, this is an interesting idea. This definitely works nicely in this case because we are always showing a list view of alerts open. Maybe for the upcoming more complex layouts this would still be fine because that business logic could be done either in the WI with the alert lists or on the client. Gonna give this a shot.

Reply:
Yeah I think it worked out nicely. Simplified the modules a lot. Let me know what you think. Approval not necessary of course because I will continue the client work in this branch but open to feedback in the meantime.